[Streams] Integrate data quality page in Streams

ElenaStoeva opened the pull request:
Closes https://github.com/elastic/kibana/issues/230258

## Summary

This PR integrates the Data quality page into Streams.

<img width="1162" height="577" alt="Screenshot 2025-09-03 at 10 42 19" src="https://github.com/user-attachments/assets/aaacceea-cee6-484b-846d-e859dc99fdf1" />

**How to test:**
1. Start Es and Kibana and load some data with `node scripts/synthtrace.js failed_logs --live --kibana=http://elastic:changeme@localhost:5601 --target=http://elastic:changeme@localhost:9200 --liveBucketSize=1000`
2. Navigate to Console and enable Streams with `POST kbn:/api/streams/_enable`
3. Navigate to Streams and click on one of the streams. Verify that the Quality tab is displayed and functions correctly.
4. Verify that the state of the Quality view is correctly saved in the URL and that loading a URL with a specific state displays the state in the view. For example, navigating to `app/streams/logs-synth.2-default/management/dataQuality?pageState=(dataStream:logs-synth.2-default,qualityIssues:(table:(page:0,rowsPerPage:25,sort:(direction:asc,field:lastOccurrence))),qualityIssuesChart:degraded,showCurrentQualityIssues:!f,timeRange:(from:now-1h,refresh:(pause:!f,value:60000),to:now),v:2)` should load the Quality tab for the `logs-synth.2-default` stream with a time range of the last 1 hour, 25 rows per page in the issues table and the table sorted by Last Occurence column.

ElenaStoeva committed:
[Streams] Integrate data quality page in Streams

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/326150)
* Commit: d28698752094a3e426b8b26947b4dacd8bd14463

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/326150#019875bc-641f-4110-a897-544b4111394d)

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326150","number":326150,"commit":"d28698752094a3e426b8b26947b4dacd8bd14463"}],"number":326150}
buildkite-pr-comment-->

ElenaStoeva committed:
Fix types and i18n

ElenaStoeva committed:
Merge branch 'main' into streams/integrate-data-quality

ElenaStoeva committed:
Merge branch 'main' into streams/integrate-data-quality

ElenaStoeva committed:
Add quality indicator badge

ElenaStoeva committed:
Merge branch 'main' into streams/integrate-data-quality

ElenaStoeva committed:
Add state to url

elasticmachine commented:
Pinging @elastic/kibana-management (Team:Kibana Management)

ElenaStoeva committed:
Merge branch 'main' into streams/integrate-data-quality

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/329794)
* Commit: dc660cc928662b062e8d5882dc0a2eae3aee720f

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/329794#0198a9a8-28d0-49f5-9a99-5d26dde885a6)

### History
* :broken_heart: [Build #329782](https://buildkite.com/elastic/kibana-pull-request/builds/329782) failed 357393a11fd238136ad81bb17b38432bceafa445
* :green_heart: [Build #326295](https://buildkite.com/elastic/kibana-pull-request/builds/326295) succeeded d288817e43d35c9278ea44436d3ea8d65a1faea4
* :broken_heart: [Build #326150](https://buildkite.com/elastic/kibana-pull-request/builds/326150) failed d28698752094a3e426b8b26947b4dacd8bd14463

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329794","number":329794,"commit":"dc660cc928662b062e8d5882dc0a2eae3aee720f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329782","number":329782,"commit":"357393a11fd238136ad81bb17b38432bceafa445"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326295","number":326295,"commit":"d288817e43d35c9278ea44436d3ea8d65a1faea4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326150","number":326150,"commit":"d28698752094a3e426b8b26947b4dacd8bd14463"}],"number":329794}
buildkite-pr-comment-->

ElenaStoeva committed:
Fix type errors

ElenaStoeva committed:
Merge branch 'streams/integrate-data-quality' of https://github.com/ElenaStoeva/kibana into streams/integrate-data-quality

ElenaStoeva committed:
Refactor

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

ElenaStoeva committed:
Reduce bundle size of datasetQuality plugin

ElenaStoeva committed:
Remove unnecessary type

ElenaStoeva committed:
Resolve merge conflicts

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

ElenaStoeva committed:
Merge branch 'main' into streams/integrate-data-quality

yngrdyn commented on x-pack/platform/plugins/shared/data_quality/public/routes/dataset_quality_details/url_schema_v1.ts:
```diff
@@ -26,6 +26,7 @@ export const getStateFromUrlValue = (
           type: 'degraded',
         }
       : undefined,
+    view: 'classic',
```
can we just skip having this exposed in `data_quality` plugin?
Instead I propose to have a default in the state, let's say `classic` and then setting that value from streams (only).
Stack management consumer doesn't need to know about these particular field

ElenaStoeva replied:
Addressed in [1659dbd](https://github.com/elastic/kibana/pull/230442/commits/1659dbdb8a7206540bfa76335a60dff51d35d5cd)

yngrdyn commented on x-pack/platform/plugins/shared/data_quality/public/routes/dataset_quality_details/url_schema_v2.ts:
```diff
@@ -21,6 +21,7 @@ export const getStateFromUrlValue = (
     showCurrentQualityIssues: urlValue.showCurrentQualityIssues,
     qualityIssuesChart: urlValue.qualityIssuesChart,
     expandedQualityIssue: urlValue.expandedQualityIssue,
+    view: 'classic',
```
same here, no need

ElenaStoeva replied:
Addressed in [1659dbd]
(https://github.com/elastic/kibana/pull/230442/commits/1659dbdb8a7206540bfa76335a60dff51d35d5cd)

yngrdyn commented on x-pack/platform/plugins/shared/streams_app/tsconfig.json:
```diff
@@ -17,6 +17,7 @@
     "@kbn/core",
     "@kbn/data-plugin",
     "@kbn/dataset-quality-plugin",
+    "@kbn/data-quality-plugin",
```
Why do we need `data-quality-plugin` as a dependency here?
With `dataset-quality-plugin` dependency should be enough

ElenaStoeva replied:
Because I was reusing some logic related to the url schema and translating it to a state, but that caused a large bundle size for the `data-quality` plugin so I moved it to a package.

afharo reviewed (approved):
`kibana.jsonc` changes LGTM

ElenaStoeva committed:
Reduce bundle size and address feedback

ElenaStoeva committed:
Merge branch 'main' into streams/integrate-data-quality

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_packages --fix'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/notice'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/334089)
* Commit: 25799b84824abaff6f5b3aa788466c87dd10741a

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/334089#0198f5f8-f381-4e99-915f-131bf46585d4)

### History
* :broken_heart: [Build #332634](https://buildkite.com/elastic/kibana-pull-request/builds/332634) failed e7a063bd51c9be0a370973d9c2f3bfa2f47d6fac
* :broken_heart: [Build #330076](https://buildkite.com/elastic/kibana-pull-request/builds/330076) failed 5061d6cc332dbcca2f993df962a0ce1d52a18fe9
* :broken_heart: [Build #330057](https://buildkite.com/elastic/kibana-pull-request/builds/330057) failed 4fa29c98f31632ad07515db06cbef207390bedf8
* :broken_heart: [Build #329961](https://buildkite.com/elastic/kibana-pull-request/builds/329961) failed 867c3d8b7a0e7f12b503104f8dd224e1a140a48c
* :broken_heart: [Build #329956](https://buildkite.com/elastic/kibana-pull-request/builds/329956) failed 157b162372d51840ef80c00197e08593e8a71a37

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334089","number":334089,"commit":"25799b84824abaff6f5b3aa788466c87dd10741a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/332634","number":332634,"commit":"e7a063bd51c9be0a370973d9c2f3bfa2f47d6fac"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330076","number":330076,"commit":"5061d6cc332dbcca2f993df962a0ce1d52a18fe9"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330057","number":330057,"commit":"4fa29c98f31632ad07515db06cbef207390bedf8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329961","number":329961,"commit":"867c3d8b7a0e7f12b503104f8dd224e1a140a48c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329956","number":329956,"commit":"157b162372d51840ef80c00197e08593e8a71a37"}],"number":334089}
buildkite-pr-comment-->

ElenaStoeva committed:
Fix quick check errors

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/334132)
* Commit: 24b6370a2b6d519e8f4aa3dc8fed7abd60ac429f

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/334132#0198f617-d829-4d9f-944a-b7ed9405250c)

### History
* :broken_heart: [Build #334089](https://buildkite.com/elastic/kibana-pull-request/builds/334089) failed 25799b84824abaff6f5b3aa788466c87dd10741a
* :broken_heart: [Build #332634](https://buildkite.com/elastic/kibana-pull-request/builds/332634) failed e7a063bd51c9be0a370973d9c2f3bfa2f47d6fac
* :broken_heart: [Build #330076](https://buildkite.com/elastic/kibana-pull-request/builds/330076) failed 5061d6cc332dbcca2f993df962a0ce1d52a18fe9
* :broken_heart: [Build #330057](https://buildkite.com/elastic/kibana-pull-request/builds/330057) failed 4fa29c98f31632ad07515db06cbef207390bedf8
* :broken_heart: [Build #329961](https://buildkite.com/elastic/kibana-pull-request/builds/329961) failed 867c3d8b7a0e7f12b503104f8dd224e1a140a48c
* :broken_heart: [Build #329956](https://buildkite.com/elastic/kibana-pull-request/builds/329956) failed 157b162372d51840ef80c00197e08593e8a71a37

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334132","number":334132,"commit":"24b6370a2b6d519e8f4aa3dc8fed7abd60ac429f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334089","number":334089,"commit":"25799b84824abaff6f5b3aa788466c87dd10741a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/332634","number":332634,"commit":"e7a063bd51c9be0a370973d9c2f3bfa2f47d6fac"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330076","number":330076,"commit":"5061d6cc332dbcca2f993df962a0ce1d52a18fe9"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330057","number":330057,"commit":"4fa29c98f31632ad07515db06cbef207390bedf8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329961","number":329961,"commit":"867c3d8b7a0e7f12b503104f8dd224e1a140a48c"}],"number":334132}
buildkite-pr-comment-->

ElenaStoeva committed:
Move badge to header

ElenaStoeva committed:
Fix type errors

kibanamachine committed:
[CI] Auto-commit changed files from 'security: 3rd-party dependencies'

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/334229)
* Commit: 298525c90236b62532c2297abe95ec9bdf03d389

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/334229#0198f6c6-291e-40ad-831b-b5b4a3560350)

### History
* :broken_heart: [Build #334132](https://buildkite.com/elastic/kibana-pull-request/builds/334132) failed 24b6370a2b6d519e8f4aa3dc8fed7abd60ac429f
* :broken_heart: [Build #334089](https://buildkite.com/elastic/kibana-pull-request/builds/334089) failed 25799b84824abaff6f5b3aa788466c87dd10741a
* :broken_heart: [Build #332634](https://buildkite.com/elastic/kibana-pull-request/builds/332634) failed e7a063bd51c9be0a370973d9c2f3bfa2f47d6fac
* :broken_heart: [Build #330076](https://buildkite.com/elastic/kibana-pull-request/builds/330076) failed 5061d6cc332dbcca2f993df962a0ce1d52a18fe9
* :broken_heart: [Build #330057](https://buildkite.com/elastic/kibana-pull-request/builds/330057) failed 4fa29c98f31632ad07515db06cbef207390bedf8
* :broken_heart: [Build #329961](https://buildkite.com/elastic/kibana-pull-request/builds/329961) failed 867c3d8b7a0e7f12b503104f8dd224e1a140a48c

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334229","number":334229,"commit":"298525c90236b62532c2297abe95ec9bdf03d389"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334132","number":334132,"commit":"24b6370a2b6d519e8f4aa3dc8fed7abd60ac429f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334089","number":334089,"commit":"25799b84824abaff6f5b3aa788466c87dd10741a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/332634","number":332634,"commit":"e7a063bd51c9be0a370973d9c2f3bfa2f47d6fac"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330076","number":330076,"commit":"5061d6cc332dbcca2f993df962a0ce1d52a18fe9"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330057","number":330057,"commit":"4fa29c98f31632ad07515db06cbef207390bedf8"}],"number":334229}
buildkite-pr-comment-->

ElenaStoeva committed:
Fix type error

ElenaStoeva committed:
Merge branch 'main' into streams/integrate-data-quality

csr reviewed (approved):
`x-pack/test/tsconfig.json` changes LGTM

ElenaStoeva committed:
Remove badge from tab in Wired

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/334568)
* Commit: 1e66a72b256a7bc59d4573888316ed18b20653d3

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/501654a602bfb45d9a21d86777a49126731d6a4b) | [after](https://github.com/elastic/kibana/commit/1e66a72b256a7bc59d4573888316ed18b20653d3) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 66 | 71 | +5 |
| `datasetQuality` | 838 | 839 | +1 |
| `streamsApp` | 617 | 636 | +19 |
| total |  |  | +25 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/501654a602bfb45d9a21d86777a49126731d6a4b) | [after](https://github.com/elastic/kibana/commit/1e66a72b256a7bc59d4573888316ed18b20653d3) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 6 | 5 | -1 |
| `datasetQuality` | 33 | 34 | +1 |
| total |  |  | -0 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/501654a602bfb45d9a21d86777a49126731d6a4b) | [after](https://github.com/elastic/kibana/commit/1e66a72b256a7bc59d4573888316ed18b20653d3) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 9.0KB | 10.1KB | +1.1KB |
| `datasetQuality` | 436.7KB | 440.9KB | +4.1KB |
| `streamsApp` | 657.4KB | 666.1KB | +8.7KB |
| total |  |  | +13.9KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/501654a602bfb45d9a21d86777a49126731d6a4b) | [after](https://github.com/elastic/kibana/commit/1e66a72b256a7bc59d4573888316ed18b20653d3) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 9.6KB | 9.8KB | +268.0B |
| `datasetQuality` | 24.5KB | 25.9KB | +1.4KB |
| `streamsApp` | 12.0KB | 12.1KB | +65.0B |
| total |  |  | +1.8KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/501654a602bfb45d9a21d86777a49126731d6a4b) | [after](https://github.com/elastic/kibana/commit/1e66a72b256a7bc59d4573888316ed18b20653d3) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 6 | 5 | -1 |
| `datasetQuality` | 33 | 34 | +1 |
| total |  |  | -0 |

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/501654a602bfb45d9a21d86777a49126731d6a4b) | [after](https://github.com/elastic/kibana/commit/1e66a72b256a7bc59d4573888316ed18b20653d3) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 16 | 18 | +2 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/501654a602bfb45d9a21d86777a49126731d6a4b) | [after](https://github.com/elastic/kibana/commit/1e66a72b256a7bc59d4573888316ed18b20653d3) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 22 | 24 | +2 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/501654a602bfb45d9a21d86777a49126731d6a4b) | [after](https://github.com/elastic/kibana/commit/1e66a72b256a7bc59d4573888316ed18b20653d3) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 24 | 26 | +2 |


</details>


### History
* :broken_heart: [Build #334229](https://buildkite.com/elastic/kibana-pull-request/builds/334229) failed 298525c90236b62532c2297abe95ec9bdf03d389
* :broken_heart: [Build #334132](https://buildkite.com/elastic/kibana-pull-request/builds/334132) failed 24b6370a2b6d519e8f4aa3dc8fed7abd60ac429f
* :broken_heart: [Build #334089](https://buildkite.com/elastic/kibana-pull-request/builds/334089) failed 25799b84824abaff6f5b3aa788466c87dd10741a
* :broken_heart: [Build #332634](https://buildkite.com/elastic/kibana-pull-request/builds/332634) failed e7a063bd51c9be0a370973d9c2f3bfa2f47d6fac
* :broken_heart: [Build #330076](https://buildkite.com/elastic/kibana-pull-request/builds/330076) failed 5061d6cc332dbcca2f993df962a0ce1d52a18fe9
* :broken_heart: [Build #330057](https://buildkite.com/elastic/kibana-pull-request/builds/330057) failed 4fa29c98f31632ad07515db06cbef207390bedf8

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334568","number":334568,"commit":"1e66a72b256a7bc59d4573888316ed18b20653d3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334229","number":334229,"commit":"298525c90236b62532c2297abe95ec9bdf03d389"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334132","number":334132,"commit":"24b6370a2b6d519e8f4aa3dc8fed7abd60ac429f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334089","number":334089,"commit":"25799b84824abaff6f5b3aa788466c87dd10741a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/332634","number":332634,"commit":"e7a063bd51c9be0a370973d9c2f3bfa2f47d6fac"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330076","number":330076,"commit":"5061d6cc332dbcca2f993df962a0ce1d52a18fe9"}],"number":334568}
buildkite-pr-comment-->

yngrdyn commented on x-pack/platform/plugins/shared/dataset_quality/public/components/quality_indicator/index.tsx:
```diff
@@ -0,0 +1,60 @@
+const OverallDataQualityIndicator = dynamic(() => import('./overall_data_quality_indicator'));
+
+interface DatasetQualityIndicatorProps {
+  controller: DatasetQualityDetailsController;
```
I don't think this is the way to go, If we want to actually expose this one as a standalone component I think we should have a separate controller, this would mean create a separate (fairly simple) machine that will be in charge on getting just thee information needed for this component to be rendered. I could use underneath the same services, endpoints, etc, but sharing the `datasetqualityDetails` state to this one imo is an overkill

yngrdyn reviewed (dismissed):
Please decouple `DatasetQualityIndicator` from the `DatasetQualityDetails`

ElenaStoeva committed:
Fix merge conflicts

ElenaStoeva committed:
Remove data quality badge in header

ElenaStoeva commented:
Hi @yngrdyn, I removed the quality badge in the header as it seems to be more complex and we need some clarifications around it from design and product. Will add it in a follow-up PR.

I also fixed the issue you mentioned with the data quality requests when not in the Data quality tab.

Could you please take another look?

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335266)
* Commit: 15a22cc6a9e201ff2545380854cfa4db0f3fd174

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/5be7a8f4c914eea310ca137407403a67fcb923e5) | [after](https://github.com/elastic/kibana/commit/15a22cc6a9e201ff2545380854cfa4db0f3fd174) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 64 | 69 | +5 |
| `streamsApp` | 617 | 634 | +17 |
| total |  |  | +22 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/5be7a8f4c914eea310ca137407403a67fcb923e5) | [after](https://github.com/elastic/kibana/commit/15a22cc6a9e201ff2545380854cfa4db0f3fd174) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 6 | 5 | -1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5be7a8f4c914eea310ca137407403a67fcb923e5) | [after](https://github.com/elastic/kibana/commit/15a22cc6a9e201ff2545380854cfa4db0f3fd174) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 8.5KB | 9.4KB | +928.0B |
| `datasetQuality` | 437.3KB | 437.6KB | +307.0B |
| `streamsApp` | 659.8KB | 667.0KB | +7.2KB |
| total |  |  | +8.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/5be7a8f4c914eea310ca137407403a67fcb923e5) | [after](https://github.com/elastic/kibana/commit/15a22cc6a9e201ff2545380854cfa4db0f3fd174) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 9.2KB | 9.4KB | +197.0B |
| `streamsApp` | 12.0KB | 12.1KB | +65.0B |
| total |  |  | +262.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/5be7a8f4c914eea310ca137407403a67fcb923e5) | [after](https://github.com/elastic/kibana/commit/15a22cc6a9e201ff2545380854cfa4db0f3fd174) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 6 | 5 | -1 |


</details>


### History
* :green_heart: [Build #335137](https://buildkite.com/elastic/kibana-pull-request/builds/335137) succeeded b9778a3972dd88720847008ca22afe12b5d351ab
* :broken_heart: [Build #335131](https://buildkite.com/elastic/kibana-pull-request/builds/335131) failed aea733411b042af8288ea98a15622913e78f0ed6
* :green_heart: [Build #334568](https://buildkite.com/elastic/kibana-pull-request/builds/334568) succeeded 1e66a72b256a7bc59d4573888316ed18b20653d3
* :broken_heart: [Build #334229](https://buildkite.com/elastic/kibana-pull-request/builds/334229) failed 298525c90236b62532c2297abe95ec9bdf03d389
* :broken_heart: [Build #334132](https://buildkite.com/elastic/kibana-pull-request/builds/334132) failed 24b6370a2b6d519e8f4aa3dc8fed7abd60ac429f
* :broken_heart: [Build #334089](https://buildkite.com/elastic/kibana-pull-request/builds/334089) failed 25799b84824abaff6f5b3aa788466c87dd10741a

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335266","number":335266,"commit":"15a22cc6a9e201ff2545380854cfa4db0f3fd174"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335137","number":335137,"commit":"b9778a3972dd88720847008ca22afe12b5d351ab"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335131","number":335131,"commit":"aea733411b042af8288ea98a15622913e78f0ed6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334568","number":334568,"commit":"1e66a72b256a7bc59d4573888316ed18b20653d3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334229","number":334229,"commit":"298525c90236b62532c2297abe95ec9bdf03d389"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334132","number":334132,"commit":"24b6370a2b6d519e8f4aa3dc8fed7abd60ac429f"}],"number":335266}
buildkite-pr-comment-->

yngrdyn reviewed (commented):
I did an overall review of the code, everything looks good
I'll perform some functional manual testing

yngrdyn commented:
The `Go to discover` is failing from a wired stream

https://github.com/user-attachments/assets/979d245b-2db6-48be-ad3d-3f049a00af37

ElenaStoeva committed:
Fix rule button bug

yngrdyn commented on x-pack/platform/plugins/shared/dataset_quality/public/components/dataset_quality_details/dataset_quality_details.tsx:
```diff
@@ -13,12 +13,16 @@ import { DataStreamNotFoundPrompt } from './index_not_found_prompt';
 // eslint-disable-next-line import/no-default-export
 export default function DatasetQualityDetails() {
+  const { datasetDetails } = useDatasetQualityDetailsState();
+  const { rawName } = datasetDetails;
```
[nit] you could use directly `dataStream`

ElenaStoeva committed:
Fix Go to discover bug

ElenaStoeva committed:
Address comment

yngrdyn commented:
There is still a problem when navigating back from discover

https://github.com/user-attachments/assets/819dd8b6-5980-47b1-adad-57e3996c2ca1

Or even if you refresh the page while being in data quality tab, the full error is

<img width="920" height="933" alt="image" src="https://github.com/user-attachments/assets/cc9b6df7-b809-466b-a6c5-e14bf522eb70" />

This is happening because we had a validator for dataStream name in `data-quality` plugin that now was moved to the package and is being reused in streams app. This validator is trying to validate dataStream name as `${type}-${dataset}-${namespace}` which is not the case for streams

ElenaStoeva committed:
Fix url to state issue

ElenaStoeva committed:
Fix merge conflicts

ElenaStoeva committed:
Fix types

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

ElenaStoeva committed:
Change approach for schema

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/336504)
* Commit: e7f45e3e4477f5b1764a2cf13d0e38c275dfcecd

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/336504#01991acf-f3c2-4e0e-a964-841efa8abc84)

### History
* :broken_heart: [Build #336380](https://buildkite.com/elastic/kibana-pull-request/builds/336380) failed db0737f543608ed2a9071f3d4d15bc2f20af38c4
* :broken_heart: [Build #336298](https://buildkite.com/elastic/kibana-pull-request/builds/336298) failed 93310f58500d4fd5993534d4b22765741a2df154
* :broken_heart: [Build #335764](https://buildkite.com/elastic/kibana-pull-request/builds/335764) failed 4564bb59241d1718fcb3303cf351858be9a40b43
* :green_heart: [Build #335266](https://buildkite.com/elastic/kibana-pull-request/builds/335266) succeeded 15a22cc6a9e201ff2545380854cfa4db0f3fd174
* :green_heart: [Build #335137](https://buildkite.com/elastic/kibana-pull-request/builds/335137) succeeded b9778a3972dd88720847008ca22afe12b5d351ab

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336504","number":336504,"commit":"e7f45e3e4477f5b1764a2cf13d0e38c275dfcecd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336380","number":336380,"commit":"db0737f543608ed2a9071f3d4d15bc2f20af38c4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336298","number":336298,"commit":"93310f58500d4fd5993534d4b22765741a2df154"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335764","number":335764,"commit":"4564bb59241d1718fcb3303cf351858be9a40b43"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335266","number":335266,"commit":"15a22cc6a9e201ff2545380854cfa4db0f3fd174"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335137","number":335137,"commit":"b9778a3972dd88720847008ca22afe12b5d351ab"}],"number":336504}
buildkite-pr-comment-->

yngrdyn commented on x-pack/platform/packages/private/kbn-data-quality/public/url_schema/dataset_quality_details/url_schema_v2.ts:
```diff
@@ -15,6 +15,7 @@ export const getStateFromUrlValue = (
 ): DatasetQualityDetailsPublicStateUpdate =>
   deepCompactObject<DatasetQualityDetailsPublicStateUpdate>({
     dataStream: urlValue.dataStream,
+    view: (urlValue as any).view,
```
```suggestion
    view: isStream(urlValue) ? urlValue.view : undefined,
```

ElenaStoeva replied:
Added with [9f6b265](https://github.com/elastic/kibana/pull/230442/commits/9f6b265ba9be900ee1a8bd89ba4c09d3bad7aefa)

yngrdyn commented on x-pack/platform/packages/private/kbn-data-quality/common/url_schema/dataset_quality_details_url_schema_v2.ts:
```diff
@@ -8,7 +8,27 @@
 import * as rt from 'io-ts';
 import { dataStreamRT, degradedFieldRT, qualityIssuesRT, timeRangeRT } from './common';
 
```
```suggestion
export const isStream = (value: unknown): value is { dataStream: string; view: 'streams' } =>
  rt.type({ dataStream: rt.string, view: rt.literal('streams') }).is(value);
```

ElenaStoeva committed:
Merge branch 'main' into streams/integrate-data-quality

ElenaStoeva committed:
Address feedback

ElenaStoeva commented on x-pack/platform/packages/private/kbn-data-quality/public/url_schema/dataset_quality_details/url_schema_v2.ts:
```diff
@@ -28,6 +29,7 @@ export const getUrlValueFromState = (
 ): datasetQualityDetailsUrlSchemaV2.UrlSchema =>
   deepCompactObject<datasetQualityDetailsUrlSchemaV2.UrlSchema>({
     dataStream: state.dataStream,
+    view: 'streams',
```
Added with [9f6b265](https://github.com/elastic/kibana/pull/230442/commits/9f6b265ba9be900ee1a8bd89ba4c09d3bad7aefa)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/337110)
* Commit: 9f6b265ba9be900ee1a8bd89ba4c09d3bad7aefa

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/337110#01992ce8-521d-4e18-ad34-fe8d76fd31e0)

### History
* :broken_heart: [Build #336504](https://buildkite.com/elastic/kibana-pull-request/builds/336504) failed e7f45e3e4477f5b1764a2cf13d0e38c275dfcecd
* :broken_heart: [Build #336380](https://buildkite.com/elastic/kibana-pull-request/builds/336380) failed db0737f543608ed2a9071f3d4d15bc2f20af38c4
* :broken_heart: [Build #336298](https://buildkite.com/elastic/kibana-pull-request/builds/336298) failed 93310f58500d4fd5993534d4b22765741a2df154
* :broken_heart: [Build #335764](https://buildkite.com/elastic/kibana-pull-request/builds/335764) failed 4564bb59241d1718fcb3303cf351858be9a40b43
* :green_heart: [Build #335266](https://buildkite.com/elastic/kibana-pull-request/builds/335266) succeeded 15a22cc6a9e201ff2545380854cfa4db0f3fd174

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337110","number":337110,"commit":"9f6b265ba9be900ee1a8bd89ba4c09d3bad7aefa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336504","number":336504,"commit":"e7f45e3e4477f5b1764a2cf13d0e38c275dfcecd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336380","number":336380,"commit":"db0737f543608ed2a9071f3d4d15bc2f20af38c4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336298","number":336298,"commit":"93310f58500d4fd5993534d4b22765741a2df154"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335764","number":335764,"commit":"4564bb59241d1718fcb3303cf351858be9a40b43"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335266","number":335266,"commit":"15a22cc6a9e201ff2545380854cfa4db0f3fd174"}],"number":337110}
buildkite-pr-comment-->

ElenaStoeva committed:
Fix mistake from merge conflict

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/337194)
* Commit: 2a8be4382bd341914593e557a065cccc6cf7d5a4

### Failed CI Steps
* [FTR Configs #107](https://buildkite.com/elastic/kibana-pull-request/builds/337194#01992dd2-5fe3-4e61-b8e9-b220a3b9c01d)
* [FTR Configs #107](https://buildkite.com/elastic/kibana-pull-request/builds/337194#01992dfc-4d16-4a17-9302-8c627157e3c9)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/337194#01992dd2-5fe3-4e61-b8e9-b220a3b9c01d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/337194/jobs/01992dd2-5fe3-4e61-b8e9-b220a3b9c01d/artifacts/01992dfc-15c9-41fb-86f1-66c108b831f9) FTR Configs #<span></span>107 / Dataset Quality Dataset Quality Details failed docs should go to discover for failed docs when the button next to breakdown selector is clicked
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/337194#01992dfc-4d16-4a17-9302-8c627157e3c9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/337194/jobs/01992dfc-4d16-4a17-9302-8c627157e3c9/artifacts/01992e05-1957-4a4d-bb73-ce1aac1ae4d3) FTR Configs #<span></span>107 / Dataset Quality Dataset Quality Details failed docs should go to discover for failed docs when the button next to breakdown selector is clicked

### History
* :broken_heart: [Build #337110](https://buildkite.com/elastic/kibana-pull-request/builds/337110) failed 9f6b265ba9be900ee1a8bd89ba4c09d3bad7aefa
* :broken_heart: [Build #336504](https://buildkite.com/elastic/kibana-pull-request/builds/336504) failed e7f45e3e4477f5b1764a2cf13d0e38c275dfcecd
* :broken_heart: [Build #336380](https://buildkite.com/elastic/kibana-pull-request/builds/336380) failed db0737f543608ed2a9071f3d4d15bc2f20af38c4
* :broken_heart: [Build #336298](https://buildkite.com/elastic/kibana-pull-request/builds/336298) failed 93310f58500d4fd5993534d4b22765741a2df154
* :broken_heart: [Build #335764](https://buildkite.com/elastic/kibana-pull-request/builds/335764) failed 4564bb59241d1718fcb3303cf351858be9a40b43

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337194","number":337194,"commit":"2a8be4382bd341914593e557a065cccc6cf7d5a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337110","number":337110,"commit":"9f6b265ba9be900ee1a8bd89ba4c09d3bad7aefa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336504","number":336504,"commit":"e7f45e3e4477f5b1764a2cf13d0e38c275dfcecd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336380","number":336380,"commit":"db0737f543608ed2a9071f3d4d15bc2f20af38c4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336298","number":336298,"commit":"93310f58500d4fd5993534d4b22765741a2df154"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335764","number":335764,"commit":"4564bb59241d1718fcb3303cf351858be9a40b43"}],"number":337194}
buildkite-pr-comment-->

ElenaStoeva committed:
Revert change to redirectLink

yngrdyn commented:
As discussed offline there is a bug on the redirection to discover but I'll fixed in a follow up PR so we can move forward with the integration

ElenaStoeva committed:
Merge branch 'main' into streams/integrate-data-quality

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/337440)
* Commit: 3c513a1b198c72cc9e69457d1a180f8481b8659b

### Failed CI Steps
* [FTR Configs #59](https://buildkite.com/elastic/kibana-pull-request/builds/337440#01992f14-fd42-4f98-887a-2f109510755b)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/337440#01992f14-fd42-4f98-887a-2f109510755b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/337440/jobs/01992f14-fd42-4f98-887a-2f109510755b/artifacts/01992f36-1a4e-4733-a9a3-1b1a4b96d6ba) FTR Configs #<span></span>59 / InfraOps App feature controls infrastructure security global infrastructure all privileges infrastructure landing page with data shows Wafflemap

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/2545c4fc76a013fd45983991a720ff587bb0dbc9) | [after](https://github.com/elastic/kibana/commit/3c513a1b198c72cc9e69457d1a180f8481b8659b) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 64 | 69 | +5 |
| `streamsApp` | 629 | 646 | +17 |
| total |  |  | +22 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/2545c4fc76a013fd45983991a720ff587bb0dbc9) | [after](https://github.com/elastic/kibana/commit/3c513a1b198c72cc9e69457d1a180f8481b8659b) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 6 | 5 | -1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/2545c4fc76a013fd45983991a720ff587bb0dbc9) | [after](https://github.com/elastic/kibana/commit/3c513a1b198c72cc9e69457d1a180f8481b8659b) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 8.5KB | 9.5KB | +1002.0B |
| `datasetQuality` | 438.6KB | 438.8KB | +235.0B |
| `streamsApp` | 665.9KB | 673.5KB | +7.7KB |
| total |  |  | +8.9KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/2545c4fc76a013fd45983991a720ff587bb0dbc9) | [after](https://github.com/elastic/kibana/commit/3c513a1b198c72cc9e69457d1a180f8481b8659b) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 9.2KB | 9.8KB | +603.0B |
| `streamsApp` | 12.0KB | 12.1KB | +65.0B |
| total |  |  | +668.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/2545c4fc76a013fd45983991a720ff587bb0dbc9) | [after](https://github.com/elastic/kibana/commit/3c513a1b198c72cc9e69457d1a180f8481b8659b) | diff |
| --- | --- | --- | --- |
| `dataQuality` | 6 | 5 | -1 |


</details>


### History
* :broken_heart: [Build #337194](https://buildkite.com/elastic/kibana-pull-request/builds/337194) failed 2a8be4382bd341914593e557a065cccc6cf7d5a4
* :broken_heart: [Build #337110](https://buildkite.com/elastic/kibana-pull-request/builds/337110) failed 9f6b265ba9be900ee1a8bd89ba4c09d3bad7aefa
* :broken_heart: [Build #336504](https://buildkite.com/elastic/kibana-pull-request/builds/336504) failed e7f45e3e4477f5b1764a2cf13d0e38c275dfcecd
* :broken_heart: [Build #336380](https://buildkite.com/elastic/kibana-pull-request/builds/336380) failed db0737f543608ed2a9071f3d4d15bc2f20af38c4
* :broken_heart: [Build #336298](https://buildkite.com/elastic/kibana-pull-request/builds/336298) failed 93310f58500d4fd5993534d4b22765741a2df154
* :broken_heart: [Build #335764](https://buildkite.com/elastic/kibana-pull-request/builds/335764) failed 4564bb59241d1718fcb3303cf351858be9a40b43

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337440","number":337440,"commit":"3c513a1b198c72cc9e69457d1a180f8481b8659b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337194","number":337194,"commit":"2a8be4382bd341914593e557a065cccc6cf7d5a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337110","number":337110,"commit":"9f6b265ba9be900ee1a8bd89ba4c09d3bad7aefa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336504","number":336504,"commit":"e7f45e3e4477f5b1764a2cf13d0e38c275dfcecd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336380","number":336380,"commit":"db0737f543608ed2a9071f3d4d15bc2f20af38c4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336298","number":336298,"commit":"93310f58500d4fd5993534d4b22765741a2df154"}],"number":337440}
buildkite-pr-comment-->

yngrdyn reviewed (approved):
Good job 🚀 
This is great
